patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -318,9 +346,20 @@
                  this.playSound(soundtype.getFallSound(), soundtype.getVolume() * 0.5F, soundtype.getPitch() * 0.75F);
              }
          }
-@@ -1675,9 +_,9 @@
+@@ -1641,7 +_,7 @@
+     protected void doHurtEquipment(DamageSource p_330843_, float p_330394_, EquipmentSlot... p_331314_) {
+         if (!(p_330394_ <= 0.0F)) {
+             int i = (int)Math.max(1.0F, p_330394_ / 4.0F);
+-
++            net.neoforged.neoforge.common.CommonHooks.onArmorHurt(p_330843_, p_331314_, p_330394_, this); p_331314_ = new EquipmentSlot[0];
```

Comment:
Do not nest multiple statements in a single line like this.  The `p_331314_ = new EquipmentSlot[0];` statement should be on its own line, with an explicit comment to note that it is for the purpose of invalidating the following for loop.